spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/BatchReadConf.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark;
+
+import java.io.Serializable;
+
+public class BatchReadConf implements Serializable {
```

Comment:
After thinking a bit more, I'd consider changing the initial approach that I suggested. How about this?

- Define separate `ParquetBatchReadConf` and `OrcBatchReadConf`, so that we can parse them separately. If there is an invalid value for a Parquet property and we are reading ORC, it should not fail the job. We can use Immutables to generate the implementations. See below.

```
@Value.Immutable
public interface ParquetBatchReadConf extends Serializable {
  ParquetReaderType readerType();

  int batchSize();
}

@Value.Immutable
public interface OrcBatchReadConf extends Serializable {
  int batchSize();
}
```

- Move all the logic to decide which reader to use and whether vectorized reads are supported to `SparkBatch`.

```
@Override
public PartitionReaderFactory createReaderFactory() {
  if (useCometBatchReads()) {
    return new SparkColumnarReaderFactory(parquetBatchReadConf(ParquetReaderType.COMET));

  } else if (useParquetBatchReads()) {
    return new SparkColumnarReaderFactory(parquetBatchReadConf(ParquetReaderType.ICEBERG));

  } else if (useOrcBatchReads()) {
    return new SparkColumnarReaderFactory(orcBatchReadConf());

  } else {
    return new SparkRowReaderFactory();
  }
}
```

```
private boolean useCometBatchReads() {
  return readConf.parquetVectorizationEnabled()
      && readConf.parquetReaderType() == ParquetReaderType.COMET
      && expectedSchema.columns().stream().allMatch(this::supportsCometBatchReads)
      && taskGroups.stream().allMatch(this::supportsParquetBatchReads);
}

private boolean supportsCometBatchReads(Types.NestedField field) {
  return field.type().isPrimitiveType() && !field.type().typeId().equals(Type.TypeID.UUID);
}
```

- Only keep `parquetReaderType()` in `SparkConf`.
- Have two fields in `BaseBatchReader`.

```
abstract class BaseBatchReader<T extends ScanTask> extends BaseReader<ColumnarBatch, T> {
  private final ParquetBatchReadConf parquetConf;
  private final OrcBatchReadConf orcConf;
```

Reply:
@aokolnychyi Thanks for the suggestion! I have made the changes.